packages/flutter/test/material/flexible_space_bar_test.dart

```diff
@@ -12,6 +12,31 @@ import 'package:flutter_test/flutter_test.dart';
 import '../widgets/semantics_tester.dart';
 
 void main() {
+  testWidgets('FlexibleSpaceBar use backgroundBuilder build background', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Scaffold(
+          appBar: AppBar(
+            toolbarHeight: 300,
+            flexibleSpace: FlexibleSpaceBar(
+              title: const Text('Title'),
+              backgroundBuilder: (BuildContext context,
+                  FlexibleSpaceBarSettings settings,
+                  BoxConstraints constraints){
+                return const Text('Background');
+              },
+              collapseMode: CollapseMode.pin,
+            ),
+          ),
+        ),
+      ),
+    );
+
+    final Finder background = find.text('Background');
+    expect(background.precache(), true);
```

Comment:
@CyJaySong this test does not validate or demonstrate how this utilizes the information provided by the settings.  The linked issue specifies being able to adapt to the stretch. This should be confirmed by the test.